src/cpp/src/block_manager.hpp

```diff
@@ -177,6 +177,15 @@ class OverwritableBlocksHashStore {
         }
         return retval;
     }
+
+    /**
+     * @brief Removes all blocks from the store
+     */
+    void clean_whole_storage() {
```

Comment:
```suggestion
    ~OverwritableBlocksHashStore() {
```
can we use dtor for such purpose?